contributors/devel/release/testing.md

```diff
@@ -0,0 +1,143 @@
+# Kubernetes test sustaining engineering
+
+This document describes how Kubernetes automated tests are maintained as part
+of the development process.
+
+## Definitions
+
+The following definitions are for tests continuously run as part of CI.
+
+- *test*
+  - *artifact*: a row in [test grid]
+  - a single test run as part of a test job
+  - maybe either an e2e test or an integration / unit test
+- *test job*
+  - *artifact*: a tab in [test grid]
+  - a collection of tests that are run together in shared environment. may:
+    - run in a specific environment - e.g. [gce, gke, aws], [cvm, gci]
+    - run under specific conditions - e.g. []upgrade, version skew, soak, serial]
+    - test a specific component - e.g. [federation, node]
+- *test infrastructure*
+  - not directly shown in the test grid
+  - libraries and infrastructure common across tests
+- *test failure*
+  - persistently failing test runs for a given test
+- *test flake*
+  - periodically failling test runs for a given test
+  
+## Ownership
+
+Each test must have an escalation point (email + slack).  The escalation point is responsible for
+keeping the test healthy.  Fixes for test failures caused by areas of ownership outside the
+responsibility of the escalation point should be coordinated with other teams by the
+test escalation point.
+
+Escalation points are expected to be responsive within 24 hours, and prioritize test failure
```

Comment:
The escalation process is TBD, and ideal end state would be to build some automation that would have a configurable and cascading escalation.  Straw man: e.g. issue -> email to SIG -> slack channel of SIG with mention SIG leads -> direct email and slack message to SIG leads

I am exploring the notion of escalating to a SIG for issues [here](https://github.com/kubernetes/community/pull/752).

Initially, escalation will be done by a human, and the contact points for SIGs are already established.  We could allow SIGs to provide a separate escalation point optionally, and default to the SIG googlegroup and slack channel.